controllers/helpers.go

```diff
@@ -256,17 +257,26 @@ func GetSubnetUUID(ctx context.Context, client *nutanixClientV3.Client, peUUID s
 		foundSubnetUUID = *subnetIntentResponse.Metadata.UUID
 	} else if subnetName != nil {
 		filter := getFilterForName(*subnetName)
-		subnetClientSideFilter := getSubnetClientSideFilter(peUUID)
-		responseSubnets, err := client.V3.ListAllSubnet(ctx, filter, subnetClientSideFilter)
+		// Not using additional filtering since we want to list overlay and vlan subnets
```

Comment:
this is a good idea specially because server side resource filtering is a bit hit & miss in my experience and doesn't always work as expected.